src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -198,9 +213,55 @@ public override async ValueTask DisposeAsync()
                     ArrayPool<char>.Shared.Return(this.buffer);
                 }
 
+                if (this.singleCharBuffer != null)
+                {
+                    ArrayPool<char>.Shared.Return(this.singleCharBuffer);
+                }
+
                 await this.FlushAsync().ConfigureAwait(false);
             }
 
+            /// <summary>
+            /// Writes the specified character to the ODataUtf8JsonWriter.
+
+            /// </summary>

```

Comment:
nit: Remove odd empty line
```suggestion
            /// Writes the specified character to the ODataUtf8JsonWriter.
            /// </summary>
```